spotify.go

```diff
@@ -107,14 +124,48 @@ func (e Error) Error() string {
 }
 
 // decodeError decodes an Error from an io.Reader.
-func decodeError(r io.Reader) error {
+func decodeError(c *Client, resp *http.Response) error {
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	buf := bytes.NewBuffer(responseBody)
+
 	var e struct {
 		E Error `json:"error"`
 	}
-	err := json.NewDecoder(r).Decode(&e)
+	err = json.NewDecoder(buf).Decode(&e)
 	if err != nil {
-		return errors.New("spotify: couldn't decode error")
+		if len(responseBody) == 0 {
+			errors.New("spotify: could not decode empty body")
+		} else {
+			return fmt.Errorf("spotify: couldn't decode error: (%d) [%s]", len(responseBody), responseBody)
+		}
 	}
+
+	if e.E.Error() == rateLimitExceededErrorMessage {
```

Comment:
I'm not crazy about comparing the error string as this could change at any time and break the library.  It looks like Spotify returns a 429 if the rate limits are exceeded - can you check this instead?